app/controllers/actions/CheckEmailVerificationFilter.scala

```diff
@@ -44,21 +46,22 @@ class CheckEmailVerificationFilterImpl(mode: Option[Mode],
     if (frontendAppConfig.emailVerificationEnabled && !mode.contains(AmendMode)) {
       request.userAnswers.get(BusinessContactDetailsPage) match {
         case Some(contactDetails) =>
-          emailVerificationService.isEmailVerified(contactDetails.emailAddress, request.userId).map {
+          emailVerificationService.isEmailVerified(contactDetails.emailAddress, request.userId).flatMap {
             case Verified =>
               logger.info("CheckEmailVerificationFilter - Verified")
-              None
+              Future(Option.empty[Result])
             case LockedTooManyLockedEmails =>
               logger.info("CheckEmailVerificationFilter - LockedTooManyLockedEmails")
-              Some(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url))
-
+              Future(Some(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url)))
             case LockedPasscodeForSingleEmail =>
               logger.info("CheckEmailVerificationFilter - LockedPasscodeForSingleEmail")
-              Some(Redirect(controllers.routes.EmailVerificationCodesExceededController.onPageLoad().url))
-
+              saveForLaterService.saveAnswers(
+                controllers.routes.EmailVerificationCodesExceededController.onPageLoad(),
+                Call(GET, request.uri)
+              )(request, executionContext, hc).map(Some(_))
             case _ =>
               logger.info("CheckEmailVerificationFilter - Not Verified")
-              Some(Redirect(controllers.routes.BusinessContactDetailsController.onPageLoad(NormalMode).url))
+              Future(Some(Redirect(controllers.routes.BusinessContactDetailsController.onPageLoad(NormalMode).url)))
```

Comment:
`Future.successful`

Reply:
Ignore Future.successful is already completed just a wrapper